include/zephyr/bluetooth/conn.h

```diff
@@ -1081,6 +1081,25 @@ void bt_conn_cb_register(struct bt_conn_cb *cb);
  */
 void bt_set_bondable(bool enable);
 
+/** @brief Overlay the bonding flag for a given connection.
+ *
+ *  Set/clear the Bonding flag in the Authentication Requirements of
+ *  SMP Pairing Request/Response data for a given connection.
+ *
+ *  The bonding flag for a given connection cannot be overlaid if
+ *  security procedures in the SMP module have already started. This function
+ *  can be called only once per connection.
+ *
+ *  If the bonding flag is not overlaid, the value will depend on global
+ *  configuration which is set using bt_set_bondable.
+ *  The default value of the global configuration is defined using
+ *  CONFIG_BT_BONDABLE Kconfig option.
+ *
+ *  @param conn Connection object.
+ *  @param enable Value allowing/disallowing to be bondable.
+ */
+int bt_bondable_overlay(struct bt_conn *conn, bool enable);
```

Comment:
I feel like `bt_conn_set_bondable()` would be the more appropriate name. I know we have `bt_smp_auth_cb_overlay()` as an internal API, but public APIs operating on a `struct bt_conn *` should preferably be prefixed with `bt_conn_`.

Reply:
Agreed, name changed.